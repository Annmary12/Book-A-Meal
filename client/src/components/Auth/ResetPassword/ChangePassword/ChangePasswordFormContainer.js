import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import * as Yup from 'yup';
import AuthForm from '../../AuthForm.js';
import { resetPassword } from '../../../../actions/resetPassword.js';

export const changePasswordFormConfig = {
  mapPropsToValues: () => ({
    password: '',
    passwordConfirm: ''
  }),
  validationSchema: Yup.object().shape({
    password: Yup.string()
      .required('Password is required')
      .min(6),
    passwordConfirm: Yup.string()
      .oneOf([Yup.ref('password'), null], "Passwords don't match!")
      .required('Password confirmation is required')
  }),
  handleSubmit: async (values, { setSubmitting, props }) => {
    setSubmitting(true);
    const resetToken = props.pathname.split('/').pop();
    const resetDetails = {
      password: values.password,
      resetToken
    };
    await props.dispatch(resetPassword(resetDetails));
    setSubmitting(false);
  }
};

const mapStateToProps = state => ({
  type: 'changepassword',
  pathname: state.router.location.pathname
});

export default compose(
  connect(mapStateToProps),
  withFormik(changePasswordFormConfig)
)(AuthForm);
