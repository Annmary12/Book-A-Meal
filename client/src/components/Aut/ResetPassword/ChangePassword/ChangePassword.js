import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import DocumentTitle from 'react-document-title';
import PropTypes from 'prop-types';
import AuthArea from '../../AuthArea.js';
import ChangePasswordFormContainer from './ChangePasswordFormContainer.js';
import checkMark from '../../../../../assets/img/check-mark.svg';


const ChangePassword = ({ resetSuccessful }) => (
  <DocumentTitle title="Change password - Book-A-Meal">
    <AuthArea>
      {
      resetSuccessful ?
        <div style={{ textAlign: 'center' }}>
          Password reset successful!
          <div style={{ marginTop: '10px' }}>
            <img width="80" src={checkMark} alt="reset-success" />
          </div>
          <p>
            <Link to="/login">Login</Link> to access your account
          </p>
        </div> :
        <Fragment>
          <ChangePasswordFormContainer />
          <div style={{ textAlign: 'center', marginTop: '10px' }}>
            <div style={{ marginBottom: '10px' }}>
              Link expired? <Link to="/reset_password">Resend link </Link>
            </div>
            <div>
              Need an account? <Link to="/signup">Sign up</Link>
            </div>
          </div>
        </Fragment>
    }
    </AuthArea>
  </DocumentTitle>
);

ChangePassword.propTypes = {
  resetSuccessful: PropTypes.bool.isRequired
};

export default ChangePassword;
