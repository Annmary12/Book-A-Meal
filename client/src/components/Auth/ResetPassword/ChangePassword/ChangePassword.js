import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import AuthArea from '../../AuthArea.js';
import ChangePasswordFormContainer from './ChangePasswordFormContainer.js';
import checkMark from '../../../../../assets/img/check-mark.svg';


const ChangePassword = ({ resetSuccessful }) => (
  <AuthArea>
    {
      resetSuccessful ?
        <div style={{ textAlign: 'center' }}>
          Password reset successful!
          <div style={{ marginTop: '10px' }}>
            <img width="80" src={checkMark} alt="reset-success" />
          </div>
          <p>
            <Link to="/login">Login</Link> to access your account
          </p>
        </div> :
        <Fragment>
          <ChangePasswordFormContainer />
          <div style={{ textAlign: 'center', marginTop: '10px' }}>
            <div style={{ marginBottom: '10px' }}>
              Link expired? <Link to="/reset_password">Resend link </Link>
            </div>
            <div>
              Need an account? <Link to="/signup">Sign up</Link>
            </div>
          </div>
        </Fragment>
    }
  </AuthArea>
);

ChangePassword.propTypes = {
  resetSuccessful: PropTypes.bool.isRequired
};

export default ChangePassword;
