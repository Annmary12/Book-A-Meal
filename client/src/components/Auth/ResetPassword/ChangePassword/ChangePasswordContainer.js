import { connect } from 'react-redux';
import ChangePassword from './ChangePassword.js';

const mapStateToProps = state => ({
  pathname: state.router.location.pathname,
  resetSuccessful: state.resetPassword.resetSuccessful
});

export default connect(mapStateToProps)(ChangePassword);
