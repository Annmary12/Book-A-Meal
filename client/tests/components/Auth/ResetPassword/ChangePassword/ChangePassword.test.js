import React from 'react';
import ChangePassword from '../../../../../src/components/Auth/ResetPassword/ChangePassword/ChangePassword';

describe('ChangePassword component', () => {
  it('should match snapshot if resetSuccessful is true', () => {
    const wrapper = shallow(
      <ChangePassword
        resetSuccessful={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if resetSuccessful is false', () => {
    const wrapper = shallow(
      <ChangePassword
        resetSuccessful={false}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});