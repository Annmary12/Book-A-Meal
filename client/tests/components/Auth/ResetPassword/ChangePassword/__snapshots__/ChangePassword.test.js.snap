// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword component should match snapshot if resetSuccessful is false 1`] = `
<AuthArea>
  <Connect(C) />
  <div
    style={
      Object {
        "marginTop": "10px",
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      Link expired? 
      <Link
        replace={false}
        to="/reset_password"
      >
        Resend link 
      </Link>
    </div>
    <div>
      Need an account? 
      <Link
        replace={false}
        to="/signup"
      >
        Sign up
      </Link>
    </div>
  </div>
</AuthArea>
`;

exports[`ChangePassword component should match snapshot if resetSuccessful is true 1`] = `
<AuthArea>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Password reset successful!
    <div
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <img
        alt="reset-success"
        src="test-file-stub"
        width="80"
      />
    </div>
    <p>
      <Link
        replace={false}
        to="/login"
      >
        Login
      </Link>
       to access your account
    </p>
  </div>
</AuthArea>
`;
